server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -303,6 +349,67 @@ static boolean execute(SearchContext searchContext,
         }
     }
 
+     private static Query tryRewriteNumericLongOrDateSort(SearchContext searchContext, IndexReader reader) throws IOException {
+        // child docs can inherit scores from parent, so avoid has_parent query
+        if (searchContext.request().source().query().getName().equals("has_parent")) return null;
+
+        if (searchContext.searchAfter() != null) return null;
+        if (searchContext.scrollContext() != null) return null;
+        if (searchContext.collapse() != null) return null;
+        Sort sort = searchContext.sort().sort;
+        if (Sort.RELEVANCE.equals(sort)) return null;
+        if (Sort.INDEXORDER.equals(sort)) return null;
```

Comment:
I think those two cases would be covered below by the `fieldName == null` check?